File: go/status_test.go
package gwp

import "testing"

func TestIsSuccess(t *testing.T) {
	if !IsSuccess(Success) {
		t.Fatal("expected success")
	}
	if IsException(Success) {
		t.Fatal("expected not exception")
	}
}

func TestOmitted(t *testing.T) {
	if !IsSuccess(OmittedResult) {
		t.Fatal("expected success for omitted")
	}
}

func TestWarning(t *testing.T) {
	if !IsWarning(Warning) {
		t.Fatal("expected warning")
	}
	if IsSuccess(Warning) {
		t.Fatal("expected not success")
	}
	if IsException(Warning) {
		t.Fatal("expected not exception")
	}
}

func TestNoData(t *testing.T) {
	if !IsNoData(NoData) {
		t.Fatal("expected no data")
	}
	if IsSuccess(NoData) {
		t.Fatal("expected not success")
	}
}

func TestException(t *testing.T) {
	if !IsException(InvalidSyntax) {
		t.Fatal("expected exception")
	}
	if IsSuccess(InvalidSyntax) {
		t.Fatal("expected not success")
	}
}

func TestGraphTypeViolation(t *testing.T) {
	if !IsException(GraphTypeViolation) {
		t.Fatal("expected exception")
	}
}

func TestStatusClass(t *testing.T) {
	tests := []struct {
		code string
		want string
	}{
		{"00000", "00"},
		{"42001", "42"},
		{"G2000", "G2"},
	}
	for _, tt := range tests {
		got := StatusClass(tt.code)
		if got != tt.want {
			t.Fatalf("StatusClass(%q) = %q, want %q", tt.code, got, tt.want)
		}
	}
}

File: js/tests/status.test.ts
import { describe, it, expect } from "vitest";
import {
  SUCCESS,
  OMITTED_RESULT,
  WARNING,
  NO_DATA,
  INVALID_SYNTAX,
  GRAPH_TYPE_VIOLATION,
  isSuccess,
  isWarning,
  isNoData,
  isException,
  statusClass,
} from "../src/status";

describe("GQLSTATUS helpers", () => {
  it("success", () => {
    expect(isSuccess(SUCCESS)).toBe(true);
    expect(isException(SUCCESS)).toBe(false);
  });

  it("omitted", () => {
    expect(isSuccess(OMITTED_RESULT)).toBe(true);
  });

  it("warning", () => {
    expect(isWarning(WARNING)).toBe(true);
    expect(isSuccess(WARNING)).toBe(false);
    expect(isException(WARNING)).toBe(false);
  });

  it("no data", () => {
    expect(isNoData(NO_DATA)).toBe(true);
    expect(isSuccess(NO_DATA)).toBe(false);
  });

  it("exception", () => {
    expect(isException(INVALID_SYNTAX)).toBe(true);
    expect(isSuccess(INVALID_SYNTAX)).toBe(false);
  });

  it("graph type violation", () => {
    expect(isException(GRAPH_TYPE_VIOLATION)).toBe(true);
  });

  it("class extraction", () => {
    expect(statusClass("00000")).toBe("00");
    expect(statusClass("42001")).toBe("42");
    expect(statusClass("G2000")).toBe("G2");
  });
});
